### ISSUE TITLE
[$1000] App doesn't open a new tab when clicking on a URL of staging on production

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22523]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to production
2. Open any report
3. Copy a deep link of staging i.e https://staging.new.expensify.com/settings/profile/display-name
4. Send the link as a message in report
5. Click on the link
## Expected Result: 
App should open a new tab to redirect to the page in staging
## Actual Result: 
App doesn't open a new tab and the link is production link

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.38-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

[Screencast from 06-07-2023 00_53_10.webm](https://github.com/Expensify/App/assets/43996225/a574322b-0f14-4c2c-bbe2-dad3c6c7880f)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dukenv0307 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688619360838289?thread_ts=1688135098.594039&cid=C049HHMV9SM

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e923e4ff221ffae9</li>
        <li>Upwork Job ID: 1678503120618921984
<li>Last Price Increase: 2023-08-07</li></li>
        2023-07-17<li>Automatic offers: </li>
<ul>
<li>situchan | Contributor | 25863058</li>
<li>dukenv0307 | Reporter | 25863059</li>
<li>Talha345 | Contributor | 26077438</li>
</ul></li>
    </ul>
</details>

**Comment by adelekennedy** [comment]
Reproduced and up to date on the Slack thread

**Comment by Talha345** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

If a staging link (staging.new.expensify.com) is sent on production app via a chat message ,the app treats it like an internal link and tries to open it internally via react-navigation instead of opening it in a new tab. 

### What is the root cause of that problem?

https://github.com/Expensify/App/blob/b105aa3d7776a164c476a982bf57939fa0ec34da/src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js#L29C3-L33

The `internalNewExpensifyPath` is only checking if the given link is an internal new expensify link i.e it is either a production or a staging link.

https://github.com/Expensify/App/blob/b105aa3d7776a164c476a982bf57939fa0ec34da/src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js#L49-L54

Since, `internalNewExpensifyPath` is true here as the given link is a staging link, so it is opened internally via react-navigation.

### What changes do you think we should make in order to solve the problem?

A new boolean variable `hasSameOrigin` should be created which checks if the given link host matches the current URL host and it should be added to the `if` condition as follows:

![image](https://github.com/Expensify/App/assets/36226639/32b94742-be33-49bd-afe1-42c7876d6cfa)

![image](https://github.com/Expensify/App/assets/36226639/b1bb22a6-55bd-4e80-9916-ae0d5d1d0682)

### What alternative solutions did you explore? (Optional)
N/A

**Comment by Talha345** [comment]
Contributor details
Your Expensify account email: talha.97.mahmood@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~0135b581d6099afe8a

**Comment by alitoshmatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
 App doesn't open a new tab when clicking on a URL of staging on production 
### What is the root cause of that problem?
The root cause is here when we are checking for the same origin we are also checking for staging url and if url matches staging url we are assuming it is the same origin 
https://github.com/Expensify/App/blob/b8aeec0bcf2cac06d51129f287f932ba0421ed4e/src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js#L28
### What changes do you think we should make in order to solve the problem?
We should remove `Url.hasSameExpensifyOrigin(attrHref, CONFIG.EXPENSIFY.STAGING_API_ROOT)` from this condition, because staging url is not the same origin as production.

We don't need to worry about opening internal urls in staging because when app is in staging environment `CONFIG.EXPENSIFY.EXPENSIFY_URL` in the condition becomes staging url 
https://github.com/Expensify/App/blob/8bc6f523aa45d970d122a9f49ed18273a6987228/.env.staging#L1 

### What alternative solutions did you explore? (Optional)

**Comment by adelekennedy** [comment]
@0xmiroslav pending proposal review

**Comment by Talha345** [comment]
@adelekennedy @0xmiroslav any update on this?

**Comment by adelekennedy** [comment]
@Talha345 we're pending @0xmiroslav review on the proposals above!

**Comment by Talha345** [comment]
@adelekennedy Yes i know but since @0xmiroslav has been MIA, I was wondering if there was a procedure in place to assign some other C+ member instead.

**Comment by Talha345** [comment]
@0xmiroslav I think STG and PROD links should be treated as external on DEV, otherwise it can be a security issue and can lead to more bugs too in the future.
Rest looks good.

**Comment by adelekennedy** [comment]
>@adelekennedy do you agree with this table?

i'm not familiar enough with DEV to comment on that @francoisl tagging you in for that one but the rest looks right to me

**Comment by francoisl** [comment]
Yes the [table here](https://github.com/Expensify/App/issues/22523#issuecomment-1638530335) makes sense to me.

**Comment by Talha345** [comment]
> Yes the [table here](https://github.com/Expensify/App/issues/22523#issuecomment-1638530335) makes sense to me.

@francoisl  @0xmiroslav what is the reasoning here to open STG and PROD links as internal on DEV?

**Comment by Talha345** [comment]
> .

I see. From this perspective it makes sense but IMHO we should keep the behaviour consistent across all ENVs and only DEV links should be treated as internal.

**Comment by francoisl** [comment]
I could see it both ways, but the more I think about it, the more I think I agree with @Talha345 that staging and production links should be external from dev, as it will make the logic easier to understand and more consistent. 

Sure, if devs want to test a link from an issue locally, they'll have to edit the domain part to `localhost:8080` to make them open internally, but that doesn't sound like a big issue to me after all.

**Comment by Talha345** [comment]
> I could see it both ways, but the more I think about it, the more I think I agree with @Talha345 that staging and production links should be external from dev, as it will make the logic easier to understand and more consistent.
> 
> Sure, if devs want to test a link from an issue locally, they'll have to edit the domain part to `localhost:8080` to make them open internally, but that doesn't sound like a big issue to me after all.

@francoisl Exactly, I think having a consistent logic here would be the best way to go forward as it would also result in more maintainable code.

Also @0xmiroslav, this also looked like a simple fix to me too, however once I had a deeper look into the logic it is not considering the case for the DEV env.

See, in the following code `attrPath` is calculated from `getURLObject` method which utilizes a regex from [here](https://github.com/Expensify/expensify-common/blob/main/lib/Url.js#L3-L5) to match the provided URL and the regex does not take into account URLs such as `localhost:8080` for DEV envs.

https://github.com/Expensify/App/blob/b8aeec0bcf2cac06d51129f287f932ba0421ed4e/src/components/HTMLEngineProvider/HTMLRenderers/AnchorRenderer.js#L23-L33

Making changes to the mentioned Regex just to accomodate localhost is not feasible because of the way it has been structured and used.

If we decide to go forward with https://github.com/Expensify/App/issues/22523#issuecomment-1641002987, https://github.com/Expensify/App/issues/22523#issuecomment-1638537890 , I can propose the following solution in addition to my [proposal](https://github.com/Expensify/App/issues/22523#issuecomment-1629939138) above: 


1. Calculation of attrPath is changed like this:

```
    const isDevUrl = attrHref.startsWith(CONST.DEV_NEW_EXPENSIFY_URL);
    const attrPath = isDevUrl ? Url.getPathFromDevURL(attrHref) : lodashGet(Url.getURLObject(attrHref), 'path', '').replace('/', '');
```
Where `getPathFromDevURL` 

```
function getPathFromDevURL(url) {
    const path = new URL(url).pathname;
    return path.substring(1); // Remove the leading '/'
}
```
2. A new simpler method to compare the origin of the provided URL with the current URL. (KISS principle)

```
function hasSameOrigin(url1, url2) {
    const host1 = new URL(url1).host;
    const host2 = new URL(url2).host;
    return host1 === host2;
}
```
This would just be used at one place like :
`const hasSameOrigin = Url.hasSameOrigin(attrHref, window.location.origin);`

3. The if clause for `internalNewExpensifyPath` is updated as follows:

```
const internalNewExpensifyPath =
        (Url.hasSameExpensifyOrigin(attrHref, CONST.NEW_EXPENSIFY_URL) || Url.hasSameExpensifyOrigin(attrHref, CONST.STAGING_NEW_EXPENSIFY_URL) || isDevUrl) &&
        !CONST.PATHS_TO_TREAT_AS_EXTERNAL.includes(attrPath)
            ? attrPath
            : '';
```

4. And finally:

```
if (internalNewExpensifyPath && hasSameOrigin) {
            Navigation.navigate(internalNewExpensifyPath);
            return;
        }
```

This would result in consistent behaviour in all environments with minimal code changes.Let me know what you think.

**Comment by Talha345** [comment]
@0xmiroslav any update?

**Comment by Talha345** [comment]
> I don't see any proposal yet which works on all platforms including native. I think we should treat localhost as internal link in dev build no matter iOS/Android.

This current functionality for this task also does not work on native apps.
For example, you go to the Android/IOS app and paste a production link.It will always open it externally which is correct in my opinion.

TL;DR This task functionality only works on web and we should keep it like this.

**Comment by Talha345** [comment]
I have never seen this behaviour anywhere.For e.g you can go to FB and post some FB link in the app and it will always open it in the browser.

Also what you said --> If native app already installed, web deep link tries to open native app.
This only works for desktop app and not on IOS/Android currently for me.

**Comment by Talha345** [comment]
> > This only works for desktop app and not on IOS/Android currently for me.
> 
> Try to play with staging/production app. This doesn't work on local build

Yes I am talking about the production Android app.

**Comment by francoisl** [comment]
> I think we should treat localhost as internal link in dev build no matter iOS/Android.


> We should stick to platform consistency. Here in this issue, we can fix "not working on native" as well.
> If native app already installed, web deep link tries to open native app. Why should this be in reverse so link in native app tries to open web?
> @francoisl do you agree?

Yes, I agree with all that. Deep links from native apps shouldn't open web.

> For example, you go to the Android/IOS app and paste a production link.It will always open it externally which is correct in my opinion.

Hm are you sure? I checked on iOS and Android, it opens the link internally for me. 

<table>
<tr>
 <td>Android
 <td>iOS (staging build, via Browserstack)
<tr>
 <td><video src="https://github.com/Expensify/App/assets/2229301/ef2f835c-d8b9-4ed8-9f89-dcbc0c178c32"/>
 <td><video src="https://github.com/Expensify/App/assets/2229301/657ecce9-8fec-4dca-9758-a3ae21e2cbf0"/>
</table>

**Comment by Talha345** [comment]
> > I think we should treat localhost as internal link in dev build no matter iOS/Android.
> 
> > We should stick to platform consistency. Here in this issue, we can fix "not working on native" as well.
> > If native app already installed, web deep link tries to open native app. Why should this be in reverse so link in native app tries to open web?
> > @francoisl do you agree?
> 
> Yes, I agree with all that. Deep links from native apps shouldn't open web.
> 
> > For example, you go to the Android/IOS app and paste a production link.It will always open it externally which is correct in my opinion.
> 
> Hm are you sure? I checked on iOS and Android, it opens the link internally for me.
> 
> Android	iOS (staging build, via Browserstack)
>  screen-20230721-171006.mp4 
>  staging_ios_links.mov

@francoisl My bad! I was simply testing with https://new.expensify.com/ and it opened in the browser and now if I test with https://new.expensify.com/settings it opens within native app.Thanks for the clarification.

Now what @0xmiroslav said makes sense.

**Comment by Talha345** [comment]
@0xmiroslav To make the solution work across all platforms,I would like to extend my solution [here](https://github.com/Expensify/App/issues/22523#issuecomment-1641109621) as follows:

Instead of using `const hasSameOrigin = Url.hasSameOrigin(attrHref, window.location.origin);` which only works on the web, I would now use `const {environmentURL} = useEnvironment();` from [here](https://github.com/Expensify/App/blob/main/src/hooks/useEnvironment.js#L6) and use it like this: `const hasSameOrigin = Url.hasSameOrigin(attrHref, environmentURL);`

I would also like to use https://www.npmjs.com/package/react-native-url-polyfill package to make the `new URL(url)` object work across all platforms since RN URL polyfill has known issues ,see https://github.com/charpeni/react-native-url-polyfill#why-do-we-need-this

Rest everything stays the same as in my above proposal.Also the methods `getPathFromDevURL` and `hasSameOrigin` in my proposal do not have guard clauses for invalid URLs which I will obviously add in the PR.


### Result
IOS Native:

https://github.com/Expensify/App/assets/36226639/a01a6ece-688c-4635-ab23-d4aca6587320

Web:


https://github.com/Expensify/App/assets/36226639/37fbb2c3-c6a9-423f-a28f-2840b20a961d

**Comment by adelekennedy** [comment]
@0xmiroslav is out for the moment - this is not a major priority so will wait until they are back!

**Comment by Talha345** [comment]
@adelekennedy @francoisl Are we going to proceed with this now?

**Comment by adelekennedy** [comment]
@Talha345 I've just assigned @situchan to review the above

**Comment by adelekennedy** [comment]
@situchan any thoughts on the above proposal?

**Comment by situchan** [comment]
I now have full context reading through all discussion. Approving today

**Comment by Talha345** [comment]
@adelekennedy @situchan I have already shared a working proposal for this issue [here](https://github.com/Expensify/App/issues/22523#issuecomment-1641109621) and an addition to it [here](https://github.com/Expensify/App/issues/22523#issuecomment-1646406206)(after some discussion on the expected behaviour) and also shared the working result videos. Why is it taking so long to approve proposals such that melvin added the Internal label now?

**Comment by adelekennedy** [comment]
I think this delay is due to switching C+ and som unavoidable delays before - I'm going to remove the internal label. @situchan I see you reviewed above, do you approve of @Talha345 proposal and we can move forward with assigning them?

**Comment by situchan** [comment]
@adelekennedy @francoisl based on discussion, can you please confirm this is correct?

| env | localhost:8082 | staging.new.expensify.com | new.expensify.com | localhost:8080
| ------------- | ------------- | ------------- | ------------- | ------------- |
| dev | internal 🔴 | external 🔴 | external 🔴 | external
| staging | external | internal | external 🔴 | external
| prod | external | external 🔴 | internal | external

On native/desktop dev build, how localhost will be treated? internal or external? I believe external?

**Comment by situchan** [comment]
Personally, I suggest to leave localhost part as is now. Only fix staging and prod

**Comment by situchan** [comment]
Another concern: what about PR build? it's env is adhoc. It's kind of staging.

When open these links in PR (i.e. #22523) build, should this be treated internal or external?
I have concerns on all platforms except web (In web, it makes sense to check if window.location.origin matches)

https://22523.pr-testing.expensify.com/ (same PR number)
https://22524.pr-testing.expensify.com/ (different PR number)
https://staging.new.expensify.com/

**Comment by situchan** [comment]
> Personally, I suggest to leave localhost part as is now. Only fix staging and prod

If localhost should be handled here as well, I suggest to increase the bounty as it's tricky

**Comment by Talha345** [comment]
> > Personally, I suggest to leave localhost part as is now. Only fix staging and prod
> 
> If localhost should be handled here as well, I suggest to increase the bounty as it's tricky

@situchan The expected behavior for this task has already been decided and it was decided to treat localhost as internal on dev on all platforms and external on STG and PROD.Similarly, on STG and PROD only links with STG and PROD respectively should be opened internally on all platforms including native.

Based on this behavior,I have already put a working proposal ,check out https://github.com/Expensify/App/issues/22523#issuecomment-1667416362 !

**Comment by situchan** [comment]
@Talha345 it still doesn't make sense to me to treat localhost:8080 as internal link while dev port is 8082

**Comment by situchan** [comment]
Also I don't see any android/iOS apps treating localhost, localhost:8080. localhost:8082 as internal links

**Comment by situchan** [comment]
> @Talha345 it still doesn't make sense to me to treat localhost:8080 as internal link while dev port is 8082

Let's say our app runs on 8082. And you run another web project on 8080.
Why deep link should be internal while real link is another app?

**Comment by Talha345** [comment]
> Also I don't see any android/iOS apps treating localhost, localhost:8080. localhost:8082 as internal links

@situchan I originally proposed the same what to leave out localhost links but it was decided with the previous C+ member and internal engineer to make behavior consistent across all environments.

Also previously, dev env was running on 8080 and now it was updated to run on 8082 but my proposal is quite generic and it would work with the new port too without any changes.

I would like to kindly request you to go through the entire conversation on this issue again.

**Comment by situchan** [comment]
I already read through full conversation. Your solution is still not acceptable.
Yet to discuss https://github.com/Expensify/App/issues/22523#issuecomment-1670765433
Also please propose entire solution in 1 comment.

**Comment by Talha345** [comment]
> > @Talha345 it still doesn't make sense to me to treat localhost:8080 as internal link while dev port is 8082
> 
> Let's say our app runs on 8082. And you run another web project on 8080.
> Why deep link should be internal while real link is another app?


If our app is running on 8082,then only the deep links with 8082 should be internal and not the ones with 8080.

**Comment by situchan** [comment]
> Okay,I will update my proposal but can you please let me know why you say that my solution is still not acceptable?

Once you updated proposal, I will re-evaluate. There's only 1 [formal proposal](https://github.com/Expensify/App/issues/22523#issuecomment-1629939138) from you and it's deprecated as you know.

**Comment by Talha345** [comment]
> const isDevUrl = attrHref.startsWith(CONST.DEV_NEW_EXPENSIFY_URL);
>     const attrPath = isDevUrl ? Url.getPathFromDevURL(attrHref) : lodashGet(Url.getURLObject(attrHref), 'path', '').replace('/', '');

I will update the original proposal but I already have shared a working proposal in the comments and have mentioned those particular comments multiple times.The proposal was extended in the comments to make the life of someone who is reading this thread easier and maintain continuity within the thread.

**Comment by situchan** [comment]
> If our app is running on 8082,then only the deep links with 8082 should be internal and not the ones with 8080.

This is super interesting case on non-web platforms.

Especially on native app, how will you treat these links?
- http://localhost
- http://localhot:8080
- http://localhost:8081
- http://localhost:8082

i.e. web port is 8082, native port is 8081

**Comment by Talha345** [comment]
@situchan proposal updated https://github.com/Expensify/App/issues/22523#issuecomment-1629939138

**Comment by Talha345** [comment]
> > If our app is running on 8082,then only the deep links with 8082 should be internal and not the ones with 8080.
> 
> This is super interesting case on non-web platforms.
> 
> Especially on native app, how will you treat these links?
> 
> * http://localhost
> * http://localhot:8080
> * http://localhost:8081
> * http://localhost:8082
> 
> i.e. web port is 8082, native port is 8081

In my proposal https://github.com/Expensify/App/issues/22523#issuecomment-1629939138 you can see that I am using `const {environmentURL} = useEnvironment();` from [here](https://github.com/Expensify/App/blob/main/src/hooks/useEnvironment.js#L6) which in turn calculates the ENV URL from [here](https://github.com/Expensify/App/blob/main/src/libs/Environment/Environment.js#L7-L12) which uses the DEV PORT from [here](https://github.com/Expensify/App/blob/main/src/CONFIG.js#L83).Therefore, whatever port this environment URL returns, only deep links with the same port will be treated as internal and others as external and this makes sense too.

**Comment by situchan** [comment]
@Talha345 can you find better solution to make it work on native without importing external package?

![Simulator Screenshot - iPhone 14 Pro - 2023-08-09 at 15 33 40](https://github.com/Expensify/App/assets/108292595/56624060-cb8a-4a4a-9584-58b95b1de608)

**Comment by Talha345** [comment]
> @Talha345 can you find better solution to make it work on native without importing external package?
> 
> ![Simulator Screenshot - iPhone 14 Pro - 2023-08-09 at 15 33 40](https://github.com/Expensify/App/assets/108292595/56624060-cb8a-4a4a-9584-58b95b1de608)
> 

The error you shared will be fixed by the external package and using the external package is already a better solution.I have also shared why we should use this package in my proposal.

**Comment by situchan** [comment]
We usually hesitate to import 3rd parties but I am fine with importing https://www.npmjs.com/package/react-native-url-polyfill given that it's popular package (400k weekly downloads) and well maintained.

**Comment by Talha345** [comment]
> We usually hesitate to import 3rd parties but I am fine with importing https://www.npmjs.com/package/react-native-url-polyfill given that it's popular package (400k weekly downloads) and well maintained.

Exactly! Since the problem this package solves is very common this package is widely used.

**Comment by situchan** [comment]
@Talha345's [proposal](https://github.com/Expensify/App/issues/22523#issuecomment-1629939138) looks good to me.
🎀 👀 🎀 C+ reviewed

**Comment by adelekennedy** [comment]
@francoisl will you give a final review to @situchan recommendation above and then I will assign @Talha345?

**Comment by situchan** [comment]
> I think we shouldn't open link internally on native/desktop platform if the hostname is localhost. That why I want to preserve the current main logic here

@tienifr I reviewed already. This is against agreed discussion. For consistency, localhost (with correct port) should be treated as internal on all platforms. Also I don't think your solution covers [this port issue](https://github.com/Expensify/App/issues/22523#issuecomment-1670788792).

Anyway, @francoisl will read through my concerns  - https://github.com/Expensify/App/issues/22523#issuecomment-1670759273, https://github.com/Expensify/App/issues/22523#issuecomment-1670765433, https://github.com/Expensify/App/issues/22523#issuecomment-1670786097 and take final decision.

**Comment by Talha345** [comment]
@situchan @francoisl  Infact I think, we can remove all the extra methods in https://github.com/Expensify/App/blob/main/src/libs/Url.js and simply use the mentioned package in my proposal to make the logic simple as follows (ofcourse with some error handling):

![Screenshot 2023-08-10 at 15 04 11](https://github.com/Expensify/App/assets/36226639/db94b94a-a12e-4ff8-8a23-02c754a28c3a)

I really do not get it how we have managed to make something as simple as comparing the origins of 2 different URLs so complex by writing 3 different methods and utilizing such a complex REGEX to make it work.I also checked the git history of the same file and saw we often have to come back and do a fix in the existing methods because of edge cases and/or errors like maximum regex stack depth reached error and so on.

**Comment by francoisl** [comment]
One thing I'm just remembering that could possibly become an issue with replacing `Url.getURLObject` is that the in the complex regex you mention, we unfortunately added some internal TLDs like `rno` or `sjc` (see https://github.com/Expensify/expensify-common/blob/main/lib/tlds.jsx), and we'd want URLs with these TLDs to keep being recognized as valid links.

<img width="511" alt="image" src="https://github.com/Expensify/App/assets/2229301/8d79f044-e0da-4e56-a9f8-89439871f212">

Would that still work with the new polyfill package?

**Comment by Talha345** [comment]
Hi @francoisl , those links should work as expected as we are not changing any code that affects which URLs are to be rendered as links or not.

The code we are changing is only being used to create a URL object from a string and then compare those URL objects origins.I also tested it with the proposed changes and everything seems to be working fine.

<img width="1509" alt="image" src="https://github.com/Expensify/App/assets/36226639/f2d8ec63-41cb-4bf3-b4ed-791818d17d84">

**Comment by francoisl** [comment]
Ah ok makes sense, thanks for clarifying. Let's go with your proposal then.

**Comment by Talha345** [comment]
Thanks @francoisl!

PR will be up tomorrow.

**Comment by Talha345** [comment]
> Ah ok makes sense, thanks for clarifying. Let's go with your proposal then.

@francoisl We have 2 options for the import of the polyfill, either import it in the root `App.js` so it overrides the default RN implementation completely or just use it where required i.e in the` Url.js` file.What do you suggest?

**Comment by Talha345** [comment]
@situchan PR is ready for review

**Comment by Talha345** [comment]
Hi @francoisl ! This PR was merged 12 days ago, any update on the payment?

**Comment by situchan** [comment]
It seems automation didn't work here. PR deployed to production on Aug 24 so payment is due tomorrow

**Comment by francoisl** [comment]
Yeah not sure what happened with the automation here, but it's due for payment now. @adelekennedy can you assist with that please?

**Comment by adelekennedy** [comment]
yikes - thanks for the heads up!

### Payouts due:

Issue Reporter: $250 @dukenv0307 
Contributor: $1000 @Talha345 (Upwork)
Contributor+: $1000 @situchan 

Eligible for 50% #urgency bonus? **N**

Upwork job is [here](https://www.upwork.com/jobs/~01e923e4ff221ffae9).

**Comment by Talha345** [comment]
> yikes - thanks for the heads up!
> 
> ### Payouts due:
> 
> Issue Reporter: $250 @dukenv0307 
> Contributor: $1000 @Talha345 (Upwork)
> Contributor+: $1000 @situchan 
> 
> Eligible for 50% #urgency bonus? **N**
> 
> Upwork job is [here](https://www.upwork.com/jobs/~01e923e4ff221ffae9).

@adelekennedy Payment received! Thanks.

Can you also leave a positive feedback comment on my profile for the Upwork job?

**Comment by ayazhussain79** [comment]
@adelekennedy @francoisl I reported regression from this bug PR please check this one https://github.com/Expensify/App/pull/24483#issuecomment-1683071524

**Comment by Talha345** [comment]
> @adelekennedy @francoisl I reported regression from this bug PR please check this one https://github.com/Expensify/App/pull/24483#issuecomment-1683071524

@ayazhussain79 it was fixed already!

**Comment by situchan** [comment]
yeah, @ayazhussain79 is eligible for reporting bonus. Thanks to their report, we were able to fix quickly before PR hit staging.

**Comment by adelekennedy** [comment]
Ah thank you! @ayazhussain79 will you apply [here](https://www.upwork.com/jobs/~01e923e4ff221ffae9)? If that kink doesn't work I may have set it to invite only

**Comment by ayazhussain79** [comment]
@adelekennedy Applied on Upwrok, Thank you

**Comment by ayazhussain79** [comment]
Offer accepted, Thank you
